exporter/clickhouseexporter/config.go

```diff
@@ -48,7 +52,14 @@ type QueueSettings struct {
 	QueueSize int `mapstructure:"queue_size"`
 }
 
+// Encapsulates ENGINE value when create table. For example, Name = ReplicatedReplacingMergeTree and Params = "'par1', 'par2', par3".
+type TableEngine struct {
+	Name string `mapstructure:"name"`
+	Params string `mapstructure:"params"`
```

Comment:
```suggestion
	Zoo_path_prefix`mapstructure:"zoo_path_prefix"`
        Replica_name: `mapstructure:"replica_name"`
        Other_parameters: `mapstructure:"other_parameters"`
```

zoo_path_prefix is the prefix of `/clickhouse/tables/{shard}/table_name`, for trace table, it will be `/clickhouse/tables/{shard}/otel_traces`.

Reply:
The idea behind TableEngine field is to be able to construct any engine with variable list of params like - AnyEngine(p1, p2, p3, p4, p5), it is not necessary have to be related to replication. Am I missing something?

Reply:
`AnyEngine(p1, p2, p3, p4, p5)` is a general idea, but we need carefully handle the zoo_path param in Replicated*MergeTree, we expect the table creation query string like:

For traces:
```
      CREATE TABLE IF NOT EXISTS otel.otel_traces on cluster 'otel' (
      ...
      ) ENGINE ReplicatedMergeTree('/clickhouse/tables/{layer}-{shard}/otel_traces', '{replica}');
```
For logs:
```
      CREATE TABLE IF NOT EXISTS otel.otel_logs on cluster 'otel' (
      ...
      ) ENGINE ReplicatedMergeTree('/clickhouse/tables/{layer}-{shard}/otel_logs', '{replica}');
```

instead of every ReplicatedMergeTree uses the same zoo_path.

Reply:
If we set `Config.TableEngine` to `TableEngine{Name: ReplicatedMergeTree}` then generated query will have cluase `ENGINE = ReplicatedMergeTree()`. Clickhouse transforms it to `ENGINE = ReplicatedMergeTree('/clickhouse/tables/{uuid}/{shard}', '{replica}')`. We can check it with `show create table otel.otel_logs`. Next we can check that placehoders actually generate unique zookeper_path for each table by executing:
```
SELECT
    fqdn(),replica_name,
    zookeeper_path,
    replica_path
FROM clusterAllReplicas('my_cluster', system.replicas)
WHERE (table = 'otel_logs') AND (database = 'otel')
```

Here is output example for `otel_traces` and `otel_logs` respectively:
```
┌─FQDN()────┬─replica_name─┬─zookeeper_path─────────────────────────────────────────────┬─replica_path───────────────────────────────────────────────────────────┐
│ localhost │ 02           │ /clickhouse/tables/052912e1-9697-4d2f-8241-4927e15062d0/01 │ /clickhouse/tables/052912e1-9697-4d2f-8241-4927e15062d0/01/replicas/02 │
└───────────┴──────────────┴────────────────────────────────────────────────────────────┴────────────────────────────────────────────────────────────────────────┘

┌─FQDN()────┬─replica_name─┬─zookeeper_path─────────────────────────────────────────────┬─replica_path───────────────────────────────────────────────────────────┐
│ localhost │ 02           │ /clickhouse/tables/8e490fbe-9857-4d25-90af-c2fffbed22b8/01 │ /clickhouse/tables/8e490fbe-9857-4d25-90af-c2fffbed22b8/01/replicas/02 │
└───────────┴──────────────┴────────────────────────────────────────────────────────────┴────────────────────────────────────────────────────────────────────────┘
```
As we can see different path is generated.
On the other hand, if user wish to provide ReplicatedMergeTree params then it is user's responsibility to do it correctly.

Reply:
thank you for your clarify